### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
This PR bumps kube machinery to `v0.25.0` and starts lock migration to coordination API. Other deps are bumped to the latest stable.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,145 +3,148 @@ module github.com/scylladb/scylla-operator
...
-	github.com/go-openapi/swag v0.19.15
-	github.com/go-openapi/validate v0.20.3
-	github.com/gocql/gocql v0.0.0-20211015133455-b225f9b53fa1
-	github.com/google/go-cmp v0.5.6
+	github.com/go-openapi/errors v0.20.3
+	github.com/go-openapi/runtime v0.24.1
+	github.com/go-openapi/strfmt v0.21.3
+	github.com/go-openapi/swag v0.22.3
+	github.com/go-openapi/validate v0.22.0
+	github.com/gocql/gocql v1.2.0
```
Comment (tnozicka) [inline go.mod:20]: @zimnx is this one ok to bump?

File: go.mod [hunk 2]
```diff
@@ -3,145 +3,148 @@ module github.com/scylladb/scylla-operator
...
+	github.com/onsi/ginkgo/v2 v2.1.5
+	github.com/onsi/gomega v1.20.1
 	github.com/pkg/errors v0.9.1
-	github.com/scylladb/go-log v0.0.6
+	github.com/scylladb/go-log v0.0.7
 	github.com/scylladb/go-set v1.0.2
-	github.com/scylladb/gocqlx/v2 v2.4.0
-	github.com/shurcooL/githubv4 v0.0.0-20210922025249-6831e00d857f
-	github.com/spf13/cobra v1.2.1
+	github.com/scylladb/gocqlx/v2 v2.7.0
```
Comment (tnozicka) [inline go.mod:31]: and here?

Comment (zimnx) [reply go.mod:31]: yes

File: go.mod (hunk 1 above)
Comment (zimnx) [reply go.mod:20]: this one shouldn't matter becasue we need to have a replace directive in go.mod. You can update our fork version though to latest one.

Comment (tnozicka) [reply go.mod:20]: right, updated

**Review by zimnx — APPROVED** [review]
LGTM, thanks

**Pull request merged**